PizzaBuilder/PizzaBuilder/IBuilder.cs

```diff
@@ -0,0 +1,145 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace PizzaBuilder
+{
+    public interface IPizzaBuilder
+    {
+        abstract void Reset();
+        void BuildDough();
+        void BuildSauce();
+        void BuildTopping(string topping);
+    }
+
+    public class WhitePizzaBuilder : IPizzaBuilder
+    {
+        private WhitePizza _product = new WhitePizza();
+        public WhitePizzaBuilder()
+        {
+            this.Reset();
+        }
+        public void Reset()
+        {
+            this._product = new WhitePizza();
+        }
+    
+        public WhitePizza GetPizza()
+        {
+            WhitePizza result = this._product;
+            this.Reset();
+            return result;
+        }
+
+        public void BuildDough()
+        {
+            this._product.Add("White Dough");
+        }
+
+        public void BuildSauce()
```

Comment:
It would be nice if you specified some sauces like (White sauce, red sauce) or what you want rather than just writing the word Sauce.